src/exr-wrap/exr-wrap.js

```diff
@@ -1,24 +1,3359 @@
+                  while (Module["postRun"].length) {
+                      addOnPostRun(Module["postRun"].shift())
+                  }
+              }
+              callRuntimeCallbacks(__ATPOSTRUN__)
+          }
+
+          function addOnPreRun(cb) {
+              __ATPRERUN__.unshift(cb)
+          }
+
+          function addOnPostRun(cb) {
+              __ATPOSTRUN__.unshift(cb)
+          }
+
+          function writeArrayToMemory(array, buffer) {
+              HEAP8.set(array, buffer)
+          }
+
+          function writeAsciiToMemory(str, buffer, dontAddNull) {
+              for (var i = 0; i < str.length; ++i) {
+                  HEAP8[buffer++ >> 0] = str.charCodeAt(i)
+              }
+              if (!dontAddNull) HEAP8[buffer >> 0] = 0
+          }
+          var runDependencies = 0;
+          var runDependencyWatcher = null;
+          var dependenciesFulfilled = null;
+
+          function addRunDependency(id) {
+              runDependencies++;
+              if (Module["monitorRunDependencies"]) {
+                  Module["monitorRunDependencies"](runDependencies)
+              }
+          }
+
+          function removeRunDependency(id) {
+              runDependencies--;
+              if (Module["monitorRunDependencies"]) {
+                  Module["monitorRunDependencies"](runDependencies)
+              }
+              if (runDependencies == 0) {
+                  if (runDependencyWatcher !== null) {
+                      clearInterval(runDependencyWatcher);
+                      runDependencyWatcher = null
+                  }
+                  if (dependenciesFulfilled) {
+                      var callback = dependenciesFulfilled;
+                      dependenciesFulfilled = null;
+                      callback()
+                  }
+              }
+          }
+          Module["preloadedImages"] = {};
+          Module["preloadedAudios"] = {};
+          var dataURIPrefix = "data:application";
+
+          function isDataURI(filename) {
+            if (typeof filename === 'string') {
+              return String.prototype.startsWith ? filename.startsWith(dataURIPrefix) : filename.indexOf(dataURIPrefix) === 0
+            }
+            return false;
+          }
+
+          function integrateWasmJS() {
+              var wasmTextFile = "exr-wrap.wast";
+              // Note that we always want to require here and not let the fallback be to load via XMLHttpRequest if it's text.
+              // The issue here is that in a parent npm project that requires jeri, if this exr-wrap.js and below requires isn't pre-processed
+              // with webpack, then the parent project doesn't know what to do with it, or we require the parent project to deal with it,
+              // which really isn't an option, therefore require the wasm and handle it here.
+              var wasmBinaryFile = require("./exr-wrap.wasm");
```

Comment:
@cbreak-black Tagging you here, it's unfortunate on the diff.  I unminified it so I could fix the webpack pieces, but then it causes the diff to appear that I changed everything, and I really didn't.  This is one line I changed, that explains why.  Without a require, webpack won't handle this piece, when webpack doesn't handle it, it causes an XMLHttpRequest both in the browser and npm.  The require let's webpack nicely handle it with the wasm-loader, which is really what we want to happen here.

Reply:
great, maybe we can check if emscripten has an option to have this done itself nowadays, or there might be a way to fix this otherwise on the generator side

Reply:
I figured out a solution I'll resolve this and tag you